### ISSUE TITLE
Swoole alpine php8.3 composer fails with a segfault 

### ISSUE THREAD

**Issue by devrla** [swoole/docker-swoole#45]
Hello, after updating the alpine php base image to v8.3.0, ext-swoole is installed and the application is running, but composer can neither install nor update.

1. What did you do? If possible, provide a simple script for reproducing the error.

```shell
docker run -it --rm phpswoole/swoole:php8.3-alpine sh
```
```shell
composer init
```

Define any dependency or try to require later, e.g.:

```shell
composer require roave/security-advisories
```

2. What did you expect to see?

Composer installation

3. What did you see instead?

`Segmentation fault`

4. What version of Swoole are you using (show your php --ri swoole)?

> Swoole => enabled
> Author => Swoole Team <team@swoole.com>
> Version => 5.1.1
> Built => Nov 27 2023 04:49:08
> coroutine => enabled with boost asm context
> epoll => enabled
> eventfd => enabled
> signalfd => enabled
> spinlock => enabled
> rwlock => enabled
> sockets => enabled
> openssl => OpenSSL 3.1.4 24 Oct 2023
> dtls => enabled
> http2 => enabled
> json => enabled
> curl-native => enabled
> pcre => enabled
> zlib => 1.2.13
> brotli => E16777225/D16777225
> mutex_timedlock => enabled
> pthread_barrier => enabled
> futex => enabled
> mysqlnd => enabled
> async_redis => enabled
> coroutine_pgsql => enabled
> 
> Directive => Local Value => Master Value
> swoole.enable_coroutine => On => On
> swoole.enable_library => On => On
> swoole.enable_fiber_mock => Off => Off
> swoole.enable_preemptive_scheduler => Off => Off
> swoole.display_errors => On => On
> swoole.use_shortname => On => On
> swoole.unixsock_buffer_size => 8388608 => 8388608
>

**Comment by liulyn** [comment]
邮件已收到 谢谢！

**Comment by deminy** [comment]
Thanks for reporting the issue. Reassigning to @matyhtf .

Steps to reproduce:

```bash
docker run --rm -ti phpswoole/swoole:5.1.1-php8.3-dev composer self-update ; echo $?
```

vs

```bash
docker run --rm -ti phpswoole/swoole:5.1.1-php8.2-dev composer self-update ; echo $?
```

**Comment by deminy** [comment]
@devrla  Yep, there could be new changes made to the curl extension between PHP 8.2 and 8.3, and the Swoole team didn't have those integrated into Swoole.

The team is aware of the issue and is working on a fix. In the meantime, removing/disabling installation option `--enable-swoole-curl` could be a temporary workaround to play with Swoole 5.1.1 on PHP 8.3, although I haven't yet tried it by myself.

Thanks

**Comment by devrla** [comment]
@deminy Thanks, the temporary workaround works for us. 👍

**Comment by devrla** [comment]
@mlocati We use the php extension installer for our images 👍

The issue has already been fixed
https://github.com/swoole/swoole-src/issues/5206

But while this test now passes...
```shell
docker run --rm -ti phpswoole/swoole:php8.3-alpine composer self-update ; echo $?
```

...a running container does not:
```shell
docker run --rm -ti phpswoole/swoole:php8.3-alpine sh
/var/www # composer self-update
Segmentation fault
```

Both behaviors probably have the same cause?

**Comment by devrla** [comment]
I have now tested all the variations from this thread again with the latest swoole alpine image. Both the composer and curl requests run without errors. The builds of our PHP 8.3 alpine images with Swoole extension are also successful. So I think the issue can be closed. Thanks to everyone who was involved in the solution.

**Comment by tiagomichaelsousa** [comment]
Hello there, is it possible to release also the changes for `5.1.1-php8.3-alpine`?

From what I saw in GitHub workflows it will probably need a merge from the main branch to the 5.1.1 branch to update the image.

Thank you in advance 🙏

**Comment by deminy** [comment]
@tiagomichaelsousa Unfortunately we can't. Docker image `5.1.1-php8.3-alpine` was built for v5.1.1 of Swoole. The issue will be fixed in the next release of Swoole, but not in the released version (v5.1.1).

However, there are [nightly images](https://github.com/swoole/docker-swoole?tab=readme-ov-file#nightly-images-built-daily-using-the-master-branch-of-swoole-src) built using latest source code of Swoole from the master branch. Please feel free to try them (e.g., [php8.3-alpine](https://github.com/swoole/docker-swoole/blob/master/dockerfiles/nightly/php8.3/alpine/Dockerfile)) a try and let us know if there are any questions.
